src/google/protobuf/stubs/atomicops.h

```diff
@@ -123,19 +123,16 @@ Atomic32 Barrier_AtomicIncrement(volatile Atomic32* ptr,
 // ensure that no later memory access can be reordered ahead of the operation.
 // "Release" operations ensure that no previous memory access can be reordered
 // after the operation.  "Barrier" operations have both "Acquire" and "Release"
-// semantics.   A MemoryBarrier() has "Barrier" semantics, but does no memory
-// access.
+// semantics.   A MemoryBarrierInternal() has "Barrier" semantics, but does no
+// memory access.
 Atomic32 Acquire_CompareAndSwap(volatile Atomic32* ptr,
                                 Atomic32 old_value,
                                 Atomic32 new_value);
 Atomic32 Release_CompareAndSwap(volatile Atomic32* ptr,
                                 Atomic32 old_value,
                                 Atomic32 new_value);
 
-#if defined(__MINGW32__) && defined(MemoryBarrier)
-#undef MemoryBarrier
-#endif
-void MemoryBarrier();
+void MemoryBarrierInternal();
```

Comment:
This is kind of a public symbol name and I would prefer not changing it. Does the original #undef not work for windows arm build? Could you try:
```cpp
#if (defined(__MINGW32__) || defined(__YOUR_PLATFORM_MACRO)) && defined(MemoryBarrier)
#undef MemoryBarrier
#endif
```
?
Or maybe simply:
```cpp
#ifdef MemoryBarrier
#undef MemoryBarrier
#endif
```